### PR TITLE
Channel switch toolbar

### PR DESCRIPTION
**Pull request by whitzhu**
NOTE: This requires toolbar hide/show code
NOTE: After MVP, want to implement closing of the menu when user clicks outside of the menu

-------CHROME-------

Switch Channel Button moved to toolbar next to the search button
screen shot 2016-08-11 at 4 38 47 pm

mobile: switch channel button hides within 'more' vertical button
screen shot 2016-08-11 at 4 38 53 pm

-------FIREFOX-------
screen shot 2016-08-11 at 4 39 10 pm

screen shot 2016-08-11 at 4 39 38 pm

screen shot 2016-08-11 at 4 49 24 pm

## Summary

*Short description*

## TODO

- [ ] Have tests been written for the new code?
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst
- [ ] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

*If you PR has a significant size, give the reviewer some helpful remarks*

## Issues addressed

List the issues solved or partly solved by the PR

## Documentation

If the PR has documentation, link the file here (either .rst in your repo or if built on Read The Docs)

## Screenshots (if appropriate)

They're nice. :)

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue [hunk 1]
```diff
@@ -1,39 +1,88 @@
...
-    </span>
-  </nav>
+  <div>
+    <nav class="nav" role="navigation" aria-label="You are here:">
+      <span class="parent full-breadcrumbs" v-if="pageMode === thePageMode.EXPLORE && pageName !== thePageNames.EXPLORE_CONTENT && !isRoot">
+        <a v-link="exploreRoot">Explore</a>
+      </span>
+      <span v-if="pageName === thePageNames.LEARN_CONTENT">
+        <a v-link="learnRoot">
+          <span class='sep'>
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue:10]: Should these still be class 'sep' ? that name doesn't seem accurate for a back arrow

File: kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue [hunk 2]
```diff
@@ -44,23 +93,58 @@
...
+    vertical-align: middle
+    fill: $core-text-annotation
+    margin-bottom: 2px
+  .full-breadcrumbs
+    @media screen and (max-width: $portrait-breakpoint)
+      display: none
+  .back
+    display: none
+    @media screen and (max-width: $portrait-breakpoint)
+      display: initial
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue:148]: please please use consistent whitespace conventions with the rest of our code :)

File: kolibri/plugins/learn/assets/src/vue/toolbar/index.vue [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        name="chan-select"
+        v-model="getCurrentChannel"
+        @change="switchChannel($event)"
+      >
+        <option v-for="channel in getChannels" :value="channel.id">{{ channel.name }}</option>
+      </select>
+    </div>
+    <search-button @scrolling="handleScroll" class='search-btn'></search-button>
+    <button class="more" @click="toggleMore" ><svg src="../icons/more-ver.svg"></svg></button>
+  </div>
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/vue/toolbar/index.vue:20]: this component looks nice and clean - nice job!

File: kolibri/plugins/learn/assets/src/vue/toolbar/index.vue [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+      },
+      /*
+      * Get the current channel ID.
+      */
+      getCurrentChannel() {
+        return this.currentChannel;
+      },
+    },
+    methods: {
+      handleScroll(position) {
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/vue/toolbar/index.vue:59]: This gets called _very_ often - would be best to put a throttle on how often the calculations below are performed.

See e.g. https://developer.mozilla.org/en-US/docs/Web/Events/scroll

Maybe cap it at 75ms? that should still feel basically instantaneous.

**Comment by indirectlylit** [comment]
This is looking really good! A few more minor updates and then we're ready to go. Thank you!

**Comment by radinamatic** [comment]
Not much to check here for keyboard navigation, since the toolbar only appears on small/mobile viewports so used by touch... :smiley: 

One thing though, @whitzhu, the channel switcher dropdown would need a `:focus` state since it's going to be navigated by keybard anyways. Check my suggestions on Rachel's PR #373!

**Comment by whitzhu** [comment]
@indirectlylit @radinamatic  

At one point in time I pushed to the wrong branch, so this stuff is in this PR as well: https://github.com/learningequality/kolibri/pull/358

So I'll close this PR, since the changes here will be in the PR above as well.

**Comment by whitzhu** [comment]
@radinamatic  Added :focus to <select> channel switch drop down menu

https://github.com/learningequality/kolibri/pull/358/commits/7da2e36b98f2257d5feb9d2a2a1ef6d74e30528b

File: kolibri/plugins/learn/assets/src/vue/toolbar/index.vue (hunk 4 above)
Comment (whitzhu) [inline kolibri/plugins/learn/assets/src/vue/toolbar/index.vue:59]: @christianmemije

**Comment by radinamatic** [comment]
@whitzhu Great, could you please include (in that PR) a small screenshot of channel switch drop down menu when in `:focus`? Thanks! :smiley:

File: kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue (hunk 2 above)
Comment (whitzhu) [inline kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue:148]: @indirectlylit Updated whitespace here: https://github.com/learningequality/kolibri/pull/358/commits/9982bdcd4321e5721995939cea0813408377fef9

File: kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue (hunk 1 above)
Comment (whitzhu) [inline kolibri/plugins/learn/assets/src/vue/breadcrumbs/index.vue:10]: @indirectlylit  Updated class name here: https://github.com/learningequality/kolibri/pull/358/commits/b7e88b8b8585cffb8a5671ebaf7ae4fe782b6f3d

**Pull request closed without merging**